### PR TITLE
[DUOS-1896][risk=no] Deprecate user properties

### PR DESCRIPTION
**Pull request by connorlbark**
## Addresses

https://broadworkbench.atlassian.net/browse/DUOS-1896

Deprecates user properties; filters out DAO calls by making user specify all the fields they want, instead of returning all possible fields.

----
Have you read [CONTRIBUTING.md](../CONTRIBUTING.md) lately? If not, do that first.

- Label PR with a Jira ticket number and include a link to the ticket
- Label PR with a security risk modifier [no, low, medium, high]
- PR describes scope of changes
- Get a minimum of one thumbs worth of review, preferably two if enough team members are available
- Get PO sign-off for all non-trivial UI or workflow changes
- Verify all tests go green
- Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: src/test/java/org/broadinstitute/consent/http/db/UserPropertyDAOTest.java
Comment (rushtong) [inline src/test/java/org/broadinstitute/consent/http/db/UserPropertyDAOTest.java:13]: I'm very surprised we have not had tests for this DAO class for so long - I missed this completely on the initial migration from mysql -> postgresql

File: src/main/java/org/broadinstitute/consent/http/db/UserPropertyDAO.java
Comment (rushtong) [inline src/main/java/org/broadinstitute/consent/http/db/UserPropertyDAO.java:24]: In addition, it looks like `findResearcherPropertiesByUserIds` is unused. Can we remove that in this PR?

File: src/main/java/org/broadinstitute/consent/http/service/ResearcherService.java
Comment (rushtong) [inline src/main/java/org/broadinstitute/consent/http/service/ResearcherService.java]: Something I like to do is add utility methods to enums to help clean up inline arguments like this. For example, if `UserFields` had this method:
```
  public static List<String> getValues() {
    return Stream.of(UserFields.values()).map(UserFields::getValue).collect(Collectors.toList());
  }
```
then this dao call could be streamlined:
```
return userPropertyDAO.findResearcherPropertiesByUser(userId, UserFields.getValues());
```

File: src/test/java/org/broadinstitute/consent/http/db/UserPropertyDAOTest.java
Comment (rushtong) [inline src/test/java/org/broadinstitute/consent/http/db/UserPropertyDAOTest.java]: Not a requirement, but consider using `RandomStringUtils` for this kind of task.

File: src/main/java/org/broadinstitute/consent/http/db/UserPropertyDAO.java
Comment (rushtong) [inline src/main/java/org/broadinstitute/consent/http/db/UserPropertyDAO.java:22]: I really like this approach, it retains the existing outdated data so we don't lose it.

File: src/main/java/org/broadinstitute/consent/http/service/ResearcherService.java
Comment (connorlbark) [reply src/main/java/org/broadinstitute/consent/http/service/ResearcherService.java]: good idea! just did it

File: src/main/java/org/broadinstitute/consent/http/db/UserPropertyDAO.java
Comment (connorlbark) [reply src/main/java/org/broadinstitute/consent/http/db/UserPropertyDAO.java:22]: thanks !!

**Comment by rushtong** [comment]
In testing this out in the UI, there is one UI bug I think we need to solve first. The DAR requires certain profile fields to be filled out with this method in `utils.js`:
```
export const completedResearcherInfoCheck = (properties) => {
  const {
    piName,
    isThePI,
    havePI,
    piEmail,
    institutionId,
  } = properties;

  const piCheck = ({isThePI, havePI, piEmail, piName}) => {
    //conditions listed are invalid checks
    //if all are true, value returned MUST be false, since pi portion of the form is incomplete
    const isThePIFalse = !evaluateTrueString(isThePI);
    const havePITrue = evaluateTrueString(havePI);
    const piAttrEmpty = isEmpty(piName) || isEmpty(piEmail);
    return !(isThePIFalse && havePITrue && piAttrEmpty);
  };

  const stringAttrs = pick(['displayName', 'address1', 'city', 'state', 'zipCode', 'country'])(properties);
  const stringAttrsCompleted = every((string) => !isEmpty(string))(stringAttrs);
  const institutionPresent = !isNil(institutionId);
  const piValid = piCheck({isThePI, havePI, piEmail, piName});
  return piValid && stringAttrsCompleted && institutionPresent;
};
```
In the new world of a reduced profile footprint, the only thing that still needs to be validated is the institution id. Can we tackle that in a companion UI PR? 

Additionally, this PR can't be merged until the updated profile page is complete. This breaks the existing profile page since it is expecting all of these fields to exist on the user.

**Review by rushtong — APPROVED** [review]
Great work, thank you! Please hold off on merging this until https://github.com/DataBiosphere/duos-ui/pull/1672 lands.

**Comment by connorlbark** [comment]
> In the new world of a reduced profile footprint, the only thing that still needs to be validated is the institution id. Can we tackle that in a companion UI PR?

got it! will add on a UI PR

**Review by quazi-h — APPROVED** [review]
Changes here look good to me. When calling the user endpoint, only the expected 5 user properties are returned. I also confirmed that the PUT/POST /api/user/profile endpoints are deprecated.

Should we also add "deprecated: true" under profileCompleted in the User yaml file? (src/main/resources/assets/schemas/User.yaml)

**Comment by connorlbark** [comment]
Good point on the swagger; deprecated that field.

Created UI PR (https://github.com/DataBiosphere/duos-ui/pull/1683) so that the frontend does not break. Will merge this once that PR is merged in.

**Pull request merged**
